### PR TITLE
BACKLOG-11740 add site switcher component to pickers

### PR DESCRIPTION
**Pull request by Taybou**

### CODE REVIEW

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 1]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                                       true
+                                   );
+                                   setIsOpen(false);
+                                   setFieldTouched(field.name, field.multiple ? [true] : true);
+                               }}
+                           >
+                               {(setSelectedItem, selectedPath, initialSelection) => {
+                                   // Build table config from picker config
+                                   /*
+                      Todo: make the picker work as CMM, use the recursionTypesFilter to browse all contents within a page
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]: A TODO should be linked to a task in jira :)

Comment (FBerthelot) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]: (even it's not your code :/ )

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 2]
```diff
@@ -31,6 +34,59 @@ const ContentPickerDialog = ({
     t,
     pickerConfig
 }) => {
+    const [site, setSite] = useState(editorContext.site);
+
+    const {data, error, loading} = useQuery(SiteNodesQuery, {
+        variables: {
+            query: 'select * from [jnt:virtualsite] where ischildnode(\'/sites\')',
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]: :o SQL in the front-end via a a graphql pipeline. I've seen everything in this world ^^

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx (hunk 1 above)
Comment (Taybou) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]: You're the owner of this code hahaha, can you please add the jira task for that?

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx (hunk 2 above)
Comment (Taybou) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]: hahaha

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx (hunk 1 above)
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 3]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                                   );
+                                   setIsOpen(false);
+                                   setFieldTouched(field.name, field.multiple ? [true] : true);
+                               }}
+                           >
+                               {(setSelectedItem, selectedPath, initialSelection) => {
+                                   // Build table config from picker config
+                                   /*
+                      Todo: make the picker work as CMM, use the recursionTypesFilter to browse all contents within a page
+                       without displaying the content lists.
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 4]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                      let isContentOrFile =
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 5]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                           selectedPath === '/sites/' + editorContext.site + '/contents' ||
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 6]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                           selectedPath.startsWith('/sites/' + editorContext.site + '/contents/') ||
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 7]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                           selectedPath === '/sites/' + editorContext.site + '/files' ||
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 8]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                                       content[0] ? content[0].id : null,
+                                       true
+                                   );
+                                   setIsOpen(false);
+                                   setFieldTouched(field.name, field.multiple ? [true] : true);
+                               }}
+                           >
+                               {(setSelectedItem, selectedPath, initialSelection) => {
+                                   // Build table config from picker config
+                                   /*
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 9]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                      Todo: make the picker work as CMM, use the recursionTypesFilter to browse all contents within a page
+                       without displaying the content lists.
+                      let isContentOrFile =
+                           selectedPath === '/sites/' + editorContext.site + '/contents' ||
+                           selectedPath.startsWith('/sites/' + editorContext.site + '/contents/') ||
+                           selectedPath === '/sites/' + editorContext.site + '/files' ||
+                           selectedPath.startsWith('/sites/' + editorContext.site + '/files/');
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 10]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 11]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                       recursionTypesFilter: isContentOrFile ? ['nt:base'] : ['jnt:page', 'jnt:contentFolder']
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx [hunk 12]
```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
...
+                       */
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx]:
```suggestion
```

**Pull request merged**